extensions/kafka/src/main/java/io/deephaven/kafka/KafkaTools.java

```diff
@@ -1529,6 +1551,10 @@ private static Table makeResultTable() {
                     ArrayBackedColumnSource.getMemoryColumnSource(int.class, null));
             resultSources.put(CONSTITUENT_COLUMN_NAME,
                     ArrayBackedColumnSource.getMemoryColumnSource(Table.class, null));
+            /*
+             * This is subtle, but this QueryTable is an inner class of StreamPartitionedTable, and so it will have a
+             * hard-ref to this, which will keep this reachable for the UpdateSourceCombiner.
+             */
```

Comment:
While the motivations to keep hard-ref are true, this does not in fact keep a hard-ref as it is written.

Reply:
Yeah, I was dumb. Fixed in the next commit.